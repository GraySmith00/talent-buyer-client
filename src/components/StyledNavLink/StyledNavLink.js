import React from 'react';
import { NavLink } from 'react-router-dom';
import './StyledNavLink.css';
import PropTypes from 'prop-types';

const StyledNavLink = ({ displayText, routeLink }) => {
  return (
    <NavLink className="nav-link" to={routeLink}>
      {displayText}
    </NavLink>
  );
};

StyledNavLink.propTypes = {
  displayText: PropTypes.string.isRequired,
  routeLink: PropTypes.string.isRequired
};

export default StyledNavLink;
