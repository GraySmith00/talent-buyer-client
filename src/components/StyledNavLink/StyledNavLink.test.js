import React from 'react';
import { shallow } from 'enzyme';

import StyledNavLink from './StyledNavLink';

describe('StyledNavLink', () => {
  let wrapper;

  beforeEach(() => {
    const mockDisplayText = 'mock';
    const displayText = mockDisplayText;
    const mockRouteLink = '/mock';
    const routeLink = mockRouteLink;

    wrapper = shallow(
      <StyledNavLink displayText={displayText} routeLink={routeLink} />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
