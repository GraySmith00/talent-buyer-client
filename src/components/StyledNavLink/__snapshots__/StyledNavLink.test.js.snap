// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledNavLink should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledNavLink
    displayText="mock"
    routeLink="/mock"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "activeClassName": "active",
      "aria-current": "page",
      "children": "mock",
      "className": "nav-link",
      "to": "/mock",
    },
    "ref": null,
    "rendered": "mock",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": "mock",
        "className": "nav-link",
        "to": "/mock",
      },
      "ref": null,
      "rendered": "mock",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
